### PR TITLE
fixed test_gpg_sign unit test

### PR DESCRIPTION
**Pull request by leifliddy**
### What does this PR do?
It fixes the `test_gpg_sign` unit test

### What issues does this PR fix or reference?
Fixes:
https://github.com/saltstack/salt/issues/62974

### Previous Behavior
It didn't correctly configure a passphrase, sign a message, or check for a signed message. 

### New Behavior
The passphrase is now correctly configured, it signs a message, and validates that a signed message was returned. 

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by leifliddy** [comment]
Ideally `sign` should return `comment/result` dict by default. 
We should add a `bare` argument (ref `encrypt` and `decrypt` functions)  
in the event where we want to return an armored signature block without the comment/res dict.  
I can modify the `sign` function to support that once this PR: https://github.com/saltstack/salt/pull/62896  
gets approved we can remove those `GPG_1_3_1` checks.   
Those version checks reference the unsupported `gnupg` library -- and it's really confusing trying to code around that.

File: changelog/62974.fixed [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed test_gpg_sign unit test
```
Comment (Ch3LL) [inline changelog/62974.fixed:1]: updates to test files do not require a changelog, only bug fixes and features. Can you remove the changelog please?

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: changelog/62974.fixed (hunk 1 above)
Comment (leifliddy) [reply changelog/62974.fixed]: Sure no prob. It's removed.

**Review by Ch3LL — APPROVED** [review]

**Comment by Ch3LL** [comment]
Merging as all tests passed before simply removing the changelog file.

**Pull request merged**